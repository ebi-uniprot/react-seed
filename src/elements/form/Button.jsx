
import React from 'react';
import PropTypes from 'prop-types';

const Button = props => (
  <button>
    { props.children }
  </button>
);

Button.propTypes = {
  children: PropTypes.node,
};

Button.defaultProps = {
  children: 'Button',
};

export default Button;
