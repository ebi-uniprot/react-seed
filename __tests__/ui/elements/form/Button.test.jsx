
import React from 'react';
import renderer from 'react-test-renderer';

import Button from '../../../../src/ui/elements/form/Button';

describe('Button component', () => {
  test('should render with a label', () => {
    const component = renderer
      .create(<Button>Works</Button>)
      .toJSON();

    expect(component).toMatchSnapshot();
  });
});
