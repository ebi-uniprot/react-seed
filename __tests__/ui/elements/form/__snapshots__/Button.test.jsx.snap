// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render with a label 1`] = `
<button>
  Works
</button>
`;
